static/src/javascripts/lib/window-performance.js

```diff
@@ -1,36 +1,9 @@
 // @flow
 
-import { noop } from 'lib/noop';
-
-const api = window.performance ||
+const api =
+    window.performance ||
     window.msPerformance ||
     window.webkitPerformance ||
-    window.mozPerformance || {
-        getEntriesByName: noop,
-
-        navigation: {
-            redirectCount: 0,
-            type: '',
-        },
-
-        mark: noop,
-        now: noop,
-
-        timing: {
-            connectStart: 0,
-            connectEnd: 0,
-
-            domainLookupEnd: 0,
-            domainLookupStart: 0,
-
-            domContentLoadedEventStart: 0,
-
-            loadEventStart: 0,
-            loadEventEnd: 0,
-
-            responseStart: 0,
-            responseEnd: 0,
-        },
-    };
+    window.mozPerformance;
```

Comment:
I would make one change... if `window.performance`, etc does not exist, then this check:

```js
if ('getEntriesByName' in performanceAPI) {
```
will throw a `ReferenceError`. A similar error will occur in [`lib/capture-perf-timings`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/lib/capture-perf-timings.js#L7-L8)

What do you think about returning `{}` by default, if `window.performance`, etc do not exist?

Reply:
I added null checks in https://github.com/guardian/frontend/pull/18110/commits/b3f90c8cde93c9cacc79ba692dca296346b9910d. Are you ok with that?